torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1700,7 +1733,15 @@ struct to_ir {
         if (actual_type->kind() == AnyType::Kind) {
           return true;
         }
-        if (auto op = actual_type->cast<OptionalType>()) {
+        if (const auto& op = actual_type->cast<UnionType>()) {
```

Comment:
When Optional inherits from Union, can we refactor this and the Optional handling ?